internal/util/managercontroller/managercontroller.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO(2.0) After converting all controllers to runttime-controller,
+// the functions in this file will no longer be needed and should be removed.
+package managercontroller
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/vmware-tanzu/velero/pkg/controller"
+)
+
+// Runnable will turn a "regular" runnable component (such as a controller)
+// into a controller-runtime Runnable
+func Runnable(p controller.Interface, numWorkers int) manager.Runnable {
+	return manager.RunnableFunc(func(stop <-chan struct{}) error {
+		ctx, cancel := contextForChannel(stop)
+		defer cancel()
+
+		return p.Run(ctx, numWorkers)
+	})
+}
+
+// contextForChannel derives a child context from a parent channel.
+//
+// The derived context's Done channel is closed when the returned cancel function
+// is called or when the parent channel is closed, whichever happens first.
+//
+// Note the caller must *always* call the CancelFunc, otherwise resources may be leaked.
+func contextForChannel(parentCh <-chan struct{}) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		select {
+		case <-parentCh:
+			cancel()
```

Comment:
Not sure if you mean instead of cancel() otherwise there's no need to add a return, that's the default behavior.

Reply:
I mean, it returns from the switch but we falls down to return ctx, cancel in the end. I don't think it's needed to add that here, this seems fine to me. If I misunderstood please let me know.

Reply:
There's no need for an explicit return - if this case is hit it will cancel the child context, exit the select statement, and then exit the goroutine.